xgi/drawing/draw.py

```diff
@@ -756,21 +757,55 @@ def _color_arg_to_dict(arg, ids, cmap):
     TypeError
         If a string, dict, iterable, or NodeStat/EdgeStat is not passed.
     """
-    if isinstance(arg, dict):
-        return {id: arg[id] for id in arg if id in ids}
-    elif isinstance(arg, str):
+
+    # single argument. Must be a string or a tuple of floats
+    if isinstance(arg, str) or (isinstance(arg, tuple) and isinstance(arg[0], float)):
         return {id: arg for id in ids}
-    elif isinstance(arg, (NodeStat, EdgeStat)):
+
+    # Iterables of colors. The values of these iterables must strings or tuples. As of now,
+    # there is not a check to verify that the tuples contain floats.
+    if isinstance(arg, Iterable):
+        if isinstance(arg, dict) and isinstance(next(iter(arg.values())), (str, tuple)):
+            return {id: arg[id] for id in arg if id in ids}
+        if isinstance(arg, (list, np.ndarray)) and isinstance(arg[0], (str, tuple)):
+            return {id: arg[idx] for idx, id in enumerate(ids)}
+
+    # Stats or iterable of values
+    if isinstance(arg, (Iterable, IDStat)):
+        # set max and min of interpolation based on color map
         if isinstance(cmap, ListedColormap):
-            vals = np.interp(arg.asnumpy(), [arg.min(), arg.max()], [0, cmap.N])
+            minval = 0
+            maxval = cmap.N
         elif isinstance(cmap, LinearSegmentedColormap):
-            vals = np.interp(arg.asnumpy(), [arg.min(), arg.max()], [0.1, 0.9])
+            minval = 0.1
+            maxval = 0.9
```

Comment:
how are these decided?

Reply:
These values are so that the colors don't fade into the background but I suppose we could add them as an arg in the future?